### ISSUE TITLE
[Proposal] review license.csv codelist

### ISSUE THREAD

**Issue by odscjen** [GFDRR/rdl-standard#123]
## What is the context or reason for the change?

The open license.csv codelist has only a few licenses covered, need to review this list and if necessary add additional commonly used licenses.

## What is your proposed change?

review and update [license.csv codelist](https://github.com/GFDRR/rdl-standard/blob/dev/codelists/open/license.csv). This is an open codelist so will not require any changes to the schema.

**Comment by pzwsk** [comment]
I would suggest to use the list of conformant licenses from https://opendefinition.org/licenses/ as baseline

**Comment by stufraser1** [comment]
THe followign are used in WB Data Catalog - we should include all of these 
![image](https://github.com/GFDRR/rdl-standard/assets/12515042/bc8c3d66-3d88-43b6-b102-ffac9badadce)

**Comment by matamadio** [comment]
Custom license to be added - that might require an additional field to specify the license

**Comment by stufraser1** [comment]
Solution - make this an open codelist?

**Comment by matamadio** [comment]
Codelist is open, main license list extented from initial version.

**Comment by duncandewhurst** [comment]
Please could you link the PR or commit(s) in which the changes were made?

**Comment by matamadio** [comment]
**Custom / Open license specified externally:** needs an additional field to enter url to custom license info.

**Comment by duncandewhurst** [comment]
The idea of the open codelist is to allow publishers to specify values that aren't in the codelist. What is the benefit of implementing a closed codelist and an additional field for values not in the codelist?

**Comment by matamadio** [comment]
Well, in those rare cases where the standard licenses are not enough i.e. custom license, providing the uri reference of the license is:
- faster then proposing a codelist update
- more complete as it provides the full license terms and not just a name+short desc
- more appropriate for custom licenses which are specific to a product - will not be reused by others

**Comment by matamadio** [comment]
Also see #281

**Comment by duncandewhurst** [comment]
Ah, I see, so the idea is to have the license URL in the data, for licenses that aren't in the license codelist. Whereas, for licenses that are in the codelist, users and tools can look up the code in the license codelist to find the URL.

That would lead to a slightly odd situation where the license URL field is only used for 'custom' licenses, or where there might be a discrepancy between the code in the `license` field and the URL in the license URL field.

Should we revisit the decision to use codes/names rather the URLs for licenses?

Looking back over the past discussions below, I can't see anything concrete about why codes/names were preferred over URLs (which would better align with DCAT in any case, per https://github.com/GFDRR/rdl-standard/issues/55#issuecomment-1553953521):

* https://github.com/GFDRR/rdl-standard/issues/55
* https://github.com/GFDRR/rdl-standard/pull/117
* https://github.com/GFDRR/rdl-standard/issues/7

**Comment by matamadio** [comment]
We can have the url field auto-populated after the license name codelist, except for "custom" where the url field appears/become editable.

**Comment by duncandewhurst** [comment]
That makes perfect sense for the MDE, but from a schema point of view, it creates a complicated set of conditions for validation (i.e. if the license name is X, the URL must be A; if the license name is Y, the URL must be B; and so on) and it means that anyone producing data via another method must implement the necessary logic to populate both fields.

I suggest that we treat the URL as the canonical identifier of a license (as recommended by DCAT) and the name as additional metadata/documentation about the URL that only belongs in the codelist, not the data. That way, the schema is kept simple, and anyone interested in producing data via a method other than the MDE has an easier implementation journey and is less likely to run into validation errors.

To be clear, that would look like this:

**Schema**

```json
{
   "license": {
      "type": "string",
      "format": "iri",
      "codelist": "license.csv",
      "openCodelist": true 
   }
}
```

**Codelist**

Code | Title | Description
--- | --- | ---
https://creativecommons.org/publicdomain/zero/1.0/ | CC0-1.0 | Creative Commons CCZero
https://creativecommons.org/licenses/by/4.0/ | CC-BY-4.0 | Creative Commons Attribution 4.0
... | ... | ...

**MDE implementation**

* Load codes (URLs), titles and/or descriptions from codelist CSV
* Populate select box with license titles or descriptions (descriptions are probably better), with a custom/add new option
* If the user selects a license from the codelist, set `license` to the associated URL from the codelist
* If the user selects custom/add new, prompt them to enter the license URL (MDE can validate that it is a valid URL) and set `license` to the entered value.

Happy to hear if you think differently.

**Comment by matamadio** [comment]
It makes sense to follow standard so I would agree on the proposal,
yet this implies to have url>title mappers for any interface where we want the text to be the license title, not the url (MDE dropdown, catalogue pages and sidebar filters).

**Comment by duncandewhurst** [comment]
>  yet this implies to have url>title mappers for any interface where we want the text to be the license title, not the url (MDE dropdown, catalogue pages and sidebar filters).

Indeed, I think that functionality is usually handled as part of the application layer (e.g. for multi-lingual interfaces) rather than the database layer (or data/schema, in our case).

**Comment by matamadio** [comment]
Alright, please make sure the codelist covers all these:
```

[
        "CC0-1.0",
        "CC-BY-4.0",
        "CC-BY-SA-4.0",
        "CC-BY-NC-SA-4.0",
        "CC BY-ND 4.0",
        "CC BY-NC-ND 4.0",
        "ODbL-1.0",
        "ODC-By-1.0",
        "PDDL-1.0",
        "Copyright",
        "Custom"
      ],
```

**Comment by stufraser1** [comment]
I agree the proposal in https://github.com/GFDRR/rdl-standard/issues/123#issuecomment-3838341341
Main things are to direct people to open licences (open definition conformant) but to be able to assign other licenses, and to give some information about that. The only other approach I can think of is to direct them to write about the license in description, which is much messier.
